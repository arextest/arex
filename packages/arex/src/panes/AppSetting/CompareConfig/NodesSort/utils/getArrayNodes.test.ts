import { describe, expect, test } from 'vitest';

import { getArrayNodes } from './getArrayNodes';

describe('getArrayNode', () => {
  test('no array object', () => {
    const object = { name: 'Helen', age: 20 };
    expect(getArrayNodes(object)).toMatchSnapshot();
  });

  test('object with simple array and sortNodeList', () => {
    const object = {
      responsestatustype: {
        responsecode: 1,
        responsedesc: 'String',
        timestamp: 1,
      },
      body: [
        {
          name: 'String',
          id: 1,
          age: 1,
        },
      ],
    };
    expect(
      getArrayNodes(
        object,
        '',
        [
          {
            id: '',
            appId: '',
            operationId: '',
            modifiedTime: 0,
            expirationType: 0,
            expirationDate: 0,
            listPath: ['body'],
            keys: [['name'], ['id'], ['age']],
            path: 'body/',
            pathKeyList: ['name/', 'id/', 'age/'],
            status: null,
          },
        ],
        'green',
      ),
    ).toMatchSnapshot();
  });

  test('object with nest array and sortNodeList', () => {
    const object = {
      responsestatustype: {
        responsecode: 1,
        responsedesc: 'String',
        timestamp: 1,
      },
      body: [
        {
          name: 'String',
          id: 1,
          age: 1,
        },
      ],
      nestArray: [
        [
          {
            name: 'String',
            id: 1,
            age: 1,
          },
        ],
      ],
    };

    expect(
      getArrayNodes(
        object,
        '',
        [
          {
            id: '',
            appId: '',
            operationId: '',
            modifiedTime: 0,
            expirationType: 0,
            expirationDate: 0,
            listPath: ['body'],
            keys: [['name'], ['id'], ['age']],
            path: 'body/',
            pathKeyList: ['name/', 'id/', 'age/'],
            status: null,
          },
          {
            id: '',
            appId: '',
            operationId: '',
            modifiedTime: 0,
            expirationType: 0,
            expirationDate: 0,
            listPath: ['nestArray'],
            keys: [['id'], ['age']],
            path: 'nestArray/',
            pathKeyList: ['id/', 'age/'],
            status: null,
          },
        ],
        'green',
      ),
    ).toMatchSnapshot();
  });
});
