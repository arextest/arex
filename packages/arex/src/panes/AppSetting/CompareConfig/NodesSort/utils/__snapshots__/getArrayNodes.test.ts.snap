// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getArrayNode > no array object 1`] = `[]`;

exports[`getArrayNode > object with nest array and sortNodeList 1`] = `
[
  {
    "children": [],
    "disabled": true,
    "icon": undefined,
    "key": "responsestatustype/",
    "title": "responsestatustype",
  },
  {
    "children": [],
    "disabled": false,
    "icon": <Badge
      color="green"
    />,
    "key": "body/",
    "title": "body",
  },
  {
    "children": [],
    "disabled": false,
    "icon": <Badge
      color="green"
    />,
    "key": "nestArray/",
    "title": "nestArray",
  },
]
`;

exports[`getArrayNode > object with simple array 1`] = `
[
  {
    "children": [],
    "disabled": true,
    "icon": undefined,
    "key": "responsestatustype/",
    "title": "responsestatustype",
  },
  {
    "children": [],
    "disabled": false,
    "icon": <Badge
      color="green"
    />,
    "key": "body/",
    "title": "body",
  },
]
`;

exports[`getArrayNode > object with simple array and sortNodeList 1`] = `
[
  {
    "children": [],
    "disabled": true,
    "icon": undefined,
    "key": "responsestatustype/",
    "title": "responsestatustype",
  },
  {
    "children": [],
    "disabled": false,
    "icon": <Badge
      color="green"
    />,
    "key": "body/",
    "title": "body",
  },
]
`;
